### ISSUE TITLE
Getting `background.service_worker` error although I'm using `background: {scripts: ...}`

### ISSUE THREAD

**Issue by mircealungu** [mozilla/web-ext#3045]
### Is this a feature request or a bug?
Bug
### What is the current behavior?

in the current folder, i have a manifest that includes: 

```
  "background": {
    "scripts": ["background.js"]
  },
```
(full manifest is [here](https://github.com/zeeguu/browser-extension/blob/8c89fa84440e2f0415689370b8df2d956aadc6b8/public/manifest.firefox.json#L34))

When I try to load the extension in Firefox from `about:debugging` or run the web-ext command I get the same error: `Error: background.service_worker is currently disabled`. However, I don't use the service_worker command so I wonder; is this a bug? Or Am I doing something wrong? I was sure that this used to work until a few weeks ago. 

```
npx web-ext run --source-dir=.

WebExtError: installTemporaryAddon: Error: Error: Could not install add-on at '/Users/gh/zeeguu/js/extension/build': Error: background.service_worker is currently disabled
    at RemoteFirefox.installTemporaryAddon (file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/lib/firefox/remote.js:92:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async FirefoxDesktopExtensionRunner.startFirefoxInstance (file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/lib/extension-runners/firefox-desktop.js:216:27)
    at async FirefoxDesktopExtensionRunner.run (file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/lib/extension-runners/firefox-desktop.js:49:5)
    at async Promise.all (index 0)
    at async MultiExtensionRunner.run (file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/lib/extension-runners/index.js:66:5)
    at async run (file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/lib/cmd/run.js:177:3)
    at async Program.execute (file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/lib/program.js:268:7)
    at async file:///Users/mircea/.npm/_npx/43d181ce0e38748d/node_modules/web-ext/bin/web-ext.js:13:1

```

Same error in the UI:

<img width="710" alt="image" src="https://github.com/mozilla/web-ext/assets/464519/649161cb-1ddb-4bf7-82af-d699eb2d5b32">

### What is the expected or desired behavior?

It used to work until recently I thought. 

### Version information (for bug reports)

- **Firefox version**: 122.0.1
- **Your OS and version**: Mac OS Sonoma 14.0.2 
- Paste the output of these commands:

```
node --version && npm --version && web-ext --version

v20.10.0
10.2.3
7.11.0

```

**Comment by mircealungu** [comment]
Hi @willdurand ! Here it is: 

```
{
  "name": "The Zeeguu Reader",
  "description": "The extension helps you read articles in the language that you want to practice and offers one-click translation and pronunciation.",
  "version": "2.2.2",
  "manifest_version": 3,
  "host_permissions": ["https://www.zeeguu.org/", "https://api.zeeguu.org/"],
  "permissions": [
    "storage",
    "activeTab",
    "scripting",
    "tabs",
    "cookies",
    "tts"
  ],

  "action": {
    "default_popup": "index.html",
    "default_title": "Open the popup",
    "default_icon": {
      "16": "/images/zeeguu16.png",
      "32": "/images/zeeguu32.png",
      "48": "/images/zeeguu48.png",
      "128": "/images/zeeguu128.png"
    }
  },
  "background": {
    "service_worker": "background.js"
  },
  "externally_connectable": {
    "matches": ["*://*.zeeguu.org/*"]
  },
  "icons": {
    "16": "/images/zeeguu16.png",
    "32": "/images/zeeguu32.png",
    "48": "/images/zeeguu48.png",
    "128": "/images/zeeguu128.png"
  },
  "web_accessible_resources": [
    {
      "resources": [
        "images/translate.svg",
        "images/sound.svg",
        "images/zeeguuLogo.svg",
        "https://zeeguu.org/static/images/trash.svg",
        "https://zeeguu.org//static/images/yellow_star",
        "https://zeeguu.org/static/images/file_rename_orange_36dp.svg",
        "https://zeeguu.org/static/images/volume_up.svg",
        "https://zeeguu.org/static/images/thumb_down_black_18dp.svg"
      ],
      "matches": ["<all_urls>"]
    }
  ]
}
```

**Comment by mircealungu** [comment]
Hi @willdurand !

This is it:

```
{
  "name": "The Zeeguu Reader",
  "description": "The extension helps you read articles in the language that you want to practice and offers one-click translation and pronunciation.",
  "version": "2.2.2",
  "manifest_version": 3,
  "host_permissions": ["https://www.zeeguu.org/", "https://api.zeeguu.org/"],
  "permissions": [
    "storage",
    "activeTab",
    "scripting",
    "tabs",
    "cookies",
    "tts"
  ],

  "action": {
    "default_popup": "index.html",
    "default_title": "Open the popup",
    "default_icon": {
      "16": "/images/zeeguu16.png",
      "32": "/images/zeeguu32.png",
      "48": "/images/zeeguu48.png",
      "128": "/images/zeeguu128.png"
    }
  },
  "background": {
    "service_worker": "background.js"
  },
  "externally_connectable": {
    "matches": ["*://*.zeeguu.org/*"]
  },
  "icons": {
    "16": "/images/zeeguu16.png",
    "32": "/images/zeeguu32.png",
    "48": "/images/zeeguu48.png",
    "128": "/images/zeeguu128.png"
  },
  "web_accessible_resources": [
    {
      "resources": [
        "images/translate.svg",
        "images/sound.svg",
        "images/zeeguuLogo.svg",
        "https://zeeguu.org/static/images/trash.svg",
        "https://zeeguu.org//static/images/yellow_star",
        "https://zeeguu.org/static/images/file_rename_orange_36dp.svg",
        "https://zeeguu.org/static/images/volume_up.svg",
        "https://zeeguu.org/static/images/thumb_down_black_18dp.svg"
      ],
      "matches": ["<all_urls>"]
    }
  ]
}

```

Wait!?! Why is there service_worker?! 
Let me double check please!

**Comment by willdurand** [comment]
Thanks. This manifest has a `background.service_worker` property so that's why you're getting an error with Firefox/web-ext. In your repo, I see your build step involves different commands, maybe one of them ends up overriding the manifest.json file that is for Firefox?

**Comment by mircealungu** [comment]
Yes. It seems that there's something broken in the build.

**Comment by mircealungu** [comment]
I found it. After some refactoring of the build commands we messed up the build system, and the chrome manifest overrides the Firefox one. I'll close this ticket now. Sorry for disturbing. Thanks for the prompt response @willdurand !

**Comment by Tgentil** [comment]
> I found it. After some refactoring of the build commands we messed up the build system, and the chrome manifest overrides the Firefox one. I'll close this ticket now. Sorry for disturbing. Thanks for the prompt response @willdurand !

@mircealungu I just had the same problem. Thank you, your issue helped me address the problem quickly.

**Comment by eladkarako** [comment]
<sub>(couple of years later...)</sub>  

tl;dr
it must also be the `manifest.json`, regardless what manifest you click when loading.  



0. 

`about:config`  
- `dom.serviceWorkers.enable` » `true`  
- `devtools.aboutdebugging.new-enabled` » `true`  


<ol><li>

```
There was an error during the temporary add-on installation.
Error details

background.service_worker is currently disabled. Add background.scripts.
```

...and make sure it is an array :|

```
There was an error during the temporary add-on installation.
Error details

Extension is invalid

Reading manifest: Error processing background.scripts: Expected array instead of "sw.js"
```

</li>
<li>

- `manifest.json` » copy to `manifest.firefox.json`  

```json
,"background"       : {"service_worker": "sw.js"}
```

to 

```json
,"background"       : {"scripts"    : ["sw.js"]
                      ,"persistent" : false
                      }
```

</li>
<li>

- `about:debugging#/runtime/this-firefox`  
- `Load Temporary Add-on…`  
- clicking `manifest.firefox.json`

still shows same error:  

```
There was an error during the temporary add-on installation.
Error details

background.service_worker is currently disabled. Add background.scripts.
```

</li>
<li>

renaming does not work. the debugger looks for `manifest.json` regardless what you are clicking...

- renaming `manifest.json` » `manifest.chrome.json`  
- renaming `manifest.firefox.json` » `manifest.json` (so weird! does it means it gets privilege to all files in that folder, is there a security risk???)  
</li>
<li>

- `about:debugging#/runtime/this-firefox`  
- `Load Temporary Add-on…`  
- clicking `manifest.json`

<h3>loads up ☺!</h3>

</li>
</ol>
